### PR TITLE
round training event corners - PBS-771

### PR DESCRIPTION
**Pull request by bmorris-revacomm**
- create View extension and custom shape to allow rounding specific corners
- refactor if statement for PersonStepperRow to round corners for first and last in group
- clean white space 😬

### CODE REVIEW

File: PBLogging/Views/TabViews/TrainingEventsTab/TrainingStepperView.swift [hunk 1]
```diff
@@ -193,42 +197,68 @@ struct PersonStepperColumn: View {
...
+                        .padding(.top, 5)
+                    Divider()
+                        .padding(.top, 8)
+                }
+                .frame(height: TrainingStepperView.rowHeight)
+                .background(
+                    color
+                        .cornerRadius(.pblCornerRadius, corners: [.topLeft, .topRight])
+                )
+            } else if personEvent.isLastInGroup {
```
Comment (jbethanco) [inline PBLogging/Views/TabViews/TrainingEventsTab/TrainingStepperView.swift:226]: Let's pair up after your appointment and figure out how to do this without the else if else

**Review by jbethanco — CHANGES_REQUESTED** [review]

File: PBLogging/Views/TabViews/TrainingEventsTab/TrainingStepperView.swift (hunk 1 above)
Comment (jbethanco) [reply PBLogging/Views/TabViews/TrainingEventsTab/TrainingStepperView.swift:226]: But I love how it looks.

Comment (Pete-Hoch) [reply PBLogging/Views/TabViews/TrainingEventsTab/TrainingStepperView.swift:226]: Agreed, it looks great.

Perhaps replace  isFirstInGroup &  isLastInGroup with roundedCorners and then the item can return none, top, bottom or something like that.

Anyway, very cool looking.

**Review by jbethanco — DISMISSED** [review]

File: PBLogging/Extensions/View+Ext.swift [hunk 2]
```diff
@@ -26,4 +26,18 @@ extension View {
...
+            return clipShape( RoundedCorner(radius: radius, corners: [.topLeft, .topRight, .bottomLeft, .bottomRight]) )
+        case (tops: true, bottoms: false):
+            return clipShape( RoundedCorner(radius: radius, corners: [.topLeft, .topRight]) )
+        case (tops: false, bottoms: true):
+            return clipShape( RoundedCorner(radius: radius, corners: [ .bottomLeft, .bottomRight]) )
+        case (tops: false, bottoms: false):
+            return clipShape( RoundedCorner(radius: radius, corners: []) )
+        }
+    }
+
```
Comment (petermisik) [inline PBLogging/Extensions/View+Ext.swift:42]: I love this extension.  Nice work.

**Review by petermisik — APPROVED** [review]
Looks fantastic.  Everything lines up and looks great!

**Pull request merged**